store/tikv/region_request.go

```diff
@@ -217,7 +221,18 @@ func (s *RegionRequestSender) SendReqCtx(
 		if err != nil {
 			return nil, nil, errors.Trace(err)
 		}
+
+		// recheck whether the session/query is killed during the Next()
+		if bo.vars != nil && bo.vars.Killed != nil && atomic.LoadUint32(bo.vars.Killed) == 1 {
```

Comment:
Maybe the only change necessary is to check `CAS(killed)` here?
And another question is, why backoff does not check it.